cmd/bosun/sched/notify.go

```diff
@@ -14,23 +14,27 @@ import (
 	"bosun.org/slog"
 )
 
-// Poll dispatches notification checks when needed.
-func (s *Schedule) Poll() {
+func (s *Schedule) dispatchNotifications() {
+	ticker := time.NewTicker(s.Conf.CheckFrequency * 2)
+	timeout := s.CheckNotifications()
 	for {
-		timeout := s.CheckNotifications()
-		// Wait for one of these two.
 		select {
 		case <-time.After(timeout):
+			timeout = s.CheckNotifications()
 		case <-s.nc:
+			timeout = s.CheckNotifications()
+		case <-ticker.C:
+			s.sendUnknownNotifications()
 		}
 	}
+
 }
 
 func (s *Schedule) Notify(st *State, n *conf.Notification) {
-	if s.notifications == nil {
-		s.notifications = make(map[*conf.Notification][]*State)
+	if s.pendingNotifications == nil {
+		s.pendingNotifications = make(map[*conf.Notification][]*State)
```

Comment:
It does require locking. Anywhere this function is called the lock should be held. We are really bad at specifying those kinds of conventions though. This should almost certainly not be exported if that is the case, and should probably note that.